### PR TITLE
Fix CI for private repos

### PR DESCRIPTION
**Pull request by vanderhe**
Sets the number of processes based on the actual resources available.

### CODE REVIEW

**Comment by vanderhe** [comment]
@bhourahine Does this fix the issue you have encountered?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -44,6 +44,13 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
 
+    - name: Get the number of available processes
+      run: |
+        NPROC=$(nproc)
+        echo "NPROC=$NPROC" >> $GITHUB_ENV
+        NPROC_HALF=$((NPROC / 2))
```
Comment (bhourahine) [inline .github/workflows/build.yml:51]: What happens if NPROC happens to be 1? Probably needs a ternary to make sure NPROC_HALF is not set to 0.

**Review by bhourahine — APPROVED** [review]
Looks OK, but one edge case to trap for.

File: .github/workflows/build.yml (hunk 1 above)
Comment (bhourahine) [reply .github/workflows/build.yml:51]: @vanderhe oh, and it ran OK for the private repo.

Comment (vanderhe) [reply .github/workflows/build.yml:51]: I assumed that GitHub never downgrades its runners to a single process, but let me deal with this edge case.

Comment (vanderhe) [reply .github/workflows/build.yml]: @bhourahine The updated version should now correctly handle the `NPROC=1` case.

**Review by bhourahine — APPROVED** [review]
LGTM

**Pull request merged**